language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/AsyncSendActionNodeContext.java

```diff
@@ -48,8 +48,10 @@ public List<LSCompletionItem> getCompletions(BallerinaCompletionContext context,
     @Override
     public void sort(BallerinaCompletionContext context, AsyncSendActionNode node, 
                      List<LSCompletionItem> completionItems) {
+        int rankOffset = 1;
         for (LSCompletionItem completionItem : completionItems) {
-            sortByAssignability(context, completionItem, SortingUtil.toRank(context, completionItem));
+            sortByAssignability(context, completionItem, SortingUtil.toRank(context, completionItem, rankOffset));
+            rankOffset++;
         }
```

Comment:
We can change the loop to iterate over the index and use index instead of rankOffset.